src/go/intelligence/int_test/test_wrapper.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/sh
+##H Script for automation of testing process of the intelligence module.
+##H Usage: test_wrapper.sh <work_directory>
+##H
+
+# If user provided the input parameter
+if [ $# -ne 1 ]; then
+    perl -ne '/^##H/ && do { s/^##H ?//; print }' <$0
+    exit 1
+fi
+
+# Setup work directory based on user input
+WDIR=$1
+TOP=$(dirname $1)
+
+if [ ! -d $WDIR ]; then
+    echo "Provided work directory '$WDIR' does not exists. creating..."
+
+    if [ ! -w $TOP ]; then
+        echo "can not create $WDIR: $USER has no write permission for '$TOP' directory. exiting."
+        exit 1
+    fi
+
+    if mkdir -p $WDIR; then
+        echo "$WDIR created !"
+    else
+        echo "Unable to create $WDIR"
+        exit 1
+    fi
+fi
+
+#Variables
+AM_BIN=$WDIR/am/alertmanager
+AM_CONFIG=$WDIR/am/alertmanager.yml
+TEST_CONFIG=$WDIR/test_config.json
+AM_VERSION="alertmanager-0.21.0.linux-amd64"
+AM_URL="https://github.com/prometheus/alertmanager/releases/download/v0.21.0/${AM_VERSION}.tar.gz"
+CMSMONITORING_REPO_URL="https://github.com/indrarahul2013/CMSMonitoring.git"
+
+PID=$(ps auxwww | egrep "alertmanager" | grep -v grep | awk 'BEGIN{ORS=" "} {print $2}')
+
+# function for starting AlertManager
+start_am() {
+    echo "Starting AlertManager in background."
+    nohup $AM_BIN --config.file=$AM_CONFIG </dev/null 2>&1 >AM.log &
+}
+
+# function for stopping AlertManager
+stop_am() {
+    local PID=$(ps auxwww | egrep "alertmanager" | grep -v grep | awk 'BEGIN{ORS=" "} {print $2}')
+    echo "Stopping Alertmanager. PID : ${PID}"
+    if [ -n "${PID}" ]; then
+        kill -9 ${PID}
+    fi
+}
+
+#function for clearing out working directory
+delete_wdir() {
+    echo "Deleting work directory : ${WDIR}"
+    rm -rf $WDIR
+}
+
+# Logic for deploying the alertmanager
+if [ ! -z "$PID" ]; then
+    echo "AlertManager already running. PID : ${PID}"
+else
+    if [ -f "$AM_BIN" ]; then
+        start_am
+    else
+        echo "AlertManager not found !!"
+        if [ -x "$(command -v wget)" ]; then
+            echo "Downloading AlertManager...."
+            wget $AM_URL -O $WDIR/am.tar.gz
```

Comment:
you can make it more generic, i.e. use either get or curl. Often wget is not installed on base Linux systems while curl is more adopted and vice versa.

Reply:
included curl too.